### PR TITLE
Arnej/add new query language features

### PR DESCRIPTION
**Pull request by arnej27959**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@kkraune please review
@geirst please review

### CODE REVIEW

**Comment by kkraune** [comment]
is the geolocation row duplicated with this?

File: documentation/reference/query-language-reference.html [hunk 1]
```diff
@@ -864,6 +923,10 @@ <h2 id="where">where</h2>
       <li> The standard <em>label</em> annotation may be used to mark the query operator with a label that can be
            referred to from the ranking expression in the rank profile. See the documentation for the
            <a href="rank-features.html#closeness(dimension,name)">closeness</a> rank feature.</li>
+      <li> The <em>distanceThreshold</em> annotation may be used to filter away hits with a higher distance
+           than the given threshold from the results.  Note that you will never get more hits with
+           <em>distanceThreshold</em> than you would get without it; if you want more hits you need to
+           increase the <em>targetHits</em> value also.
```
Comment (geirst) [inline documentation/reference/query-language-reference.html:929]: Perhaps we should add a note that this threshold must correspond to the range of values output by the distance metric specified (with link to "schema-reference.html#distance-metric")?

File: documentation/reference/query-language-reference.html [hunk 2]
```diff
@@ -805,6 +805,65 @@ <h2 id="where">where</h2>
     </table>
   </td></tr>
 
+  <tr id="geoLocation"><th>geoLocation</th><td>
```
Comment (geirst) [inline documentation/reference/query-language-reference.html:808]: This is already documented: https://docs.vespa.ai/documentation/reference/query-language-reference.html#geoLocation, and duplicates the row above.

**Comment by arnej27959** [comment]
seems I didn't make my branch correctly, sorry about that.
Please take another look now.

**Review by geirst — APPROVED** [review]
👍

**Review by kkraune — COMMENTED** [review]
👍

**Pull request merged**